### PR TITLE
Feature/logging on `simulation_objects`

### LINKED ISSUE #25: Implement logging on the correct methods
## User stories
As a data analyst I want complete sample data in order to gain correct insights into the simulation
 
## Dependencies
Depends on: 
 - #16 
 - #20 

## (Wiki) References
- <https://github.com/BP2022-AP1/bp2022-ap1/wiki/Logging>

## Acceptance Criteria
- [x] Train spawn creates log event
- [x] Train removal creates log event
- [x] Train arrival creates log event
- [x] Train departure creates log event
- [x] Fahrstrassen forming creates log event
- [x] Fahrstrassen dissolving creates log event
- [x] Signal switching creates log event
- [x] (needs discussion) fault injection creates log event
- [x] (needs discussion) fault removal creates log event

### PR DESCRIPTION
**Pull request by SaturnHafen**
Fixes #25 

I added the event bus calls in the corresponding places and tested them using mocks.

I merged #538 already to get the (new) timetable logic.

## PR checklist

- [x] Acceptance criteria fulfilled
- [x] Additional features are tested
- [ ] ~~Docs (code, wiki & diagrams) updated~~
- [ ] ~~Breaking changes anounced~~
- [ ] ~~Added classes that inherit from `BaseModel` to `src.constants.tables`~~
- [ ] Dev-branch has been merged into local branch to resolve conflicts
- [ ] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

**Review by antonykamp — APPROVED** [review]
I'm sorry for the long delay 😄 I forgot it somehow.
It's a big PR with many changes, but it looks fine to me 👍 
Thanks for your work 💚

**Comment by SaturnHafen** [comment]
The failing tests are something @Lietze has added. I don't know how to fix them.

File: src/interlocking_component/route_controller.py [hunk 1]
```diff
@@ -307,6 +307,23 @@ def maybe_set_fahrstrasse(self, train: Train, edge: Edge):
...
+                    == train.reserved_tracks[1].reservations[0][1].identifier
+                )
+        except Exception as exc:
+            for track in train.reserved_tracks:
+                print(track.edges[0].identifier)
+            raise ValueError(
+                ("Something is off: Track segment " f"{train.edge.identifier}")
+            ) from exc
+        train.edge.track.reservations.pop(0)
+        train.reserved_tracks.pop(0)
```
Comment (Lietze) [inline src/interlocking_component/route_controller.py]: Why is this in this PR? By now is is handled in the infrastructure provider which calls the RouteController.

**Review by Lietze — CHANGES_REQUESTED** [review]
Hey this looks good now, but I think one more change is needed

**Review by Lietze — APPROVED** [review]
Good job

**Pull request merged**